### PR TITLE
Add Data Source Validation

### PR DESCRIPTION
**Pull request by gitsierra**
## Purpose

https://qmacbis.atlassian.net/browse/OY2-17347

Application endpoint:  https://d35zilecflx1gj.cloudfront.net/

Endpoint: TBD

#### Pull Request Creator Checklist

- [x] This PR has an associated issue or issues.
- [x] The associated issue(s) are linked above.
- [x] This PR meets all acceptance criteria for those issues.
- [x] This PR and linked issue(s) are adequately documented
- [x] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.
- [x] Someone has been assigned this PR.
- [x] At least one person has been marked as reviewer on this PR.

#### Pull Request Reviewer/Assignee Checklist

- [ ] This PR has an associated issue or issues.
- [ ] The associated issue(s) are linked above.
- [ ] This PR meets all acceptance criteria for those issues.
- [ ] This PR and linked issue(s) are adequately documented
- [ ] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.

### CODE REVIEW

**Comment by abelzer212** [comment]
It looks good, it looks like you missed the last batch of measures that went in yesterday though: AMB, WCC, and 3 others.

**Review by abelzer212 — DISMISSED** [review]

File: services/ui-src/src/measures/globalValidations/tests/validateOneDataSource.test.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  const _check_errors = (data: any, numErrors: number) => {
+    errorArray = [...validateOneDataSource(data)];
+    expect(errorArray.length).toBe(numErrors);
+  };
+
+  beforeEach(() => {
+    formData = JSON.parse(JSON.stringify(testFormData)); // reset data
+    errorArray = [];
+  });
+
```
Comment (jphechter) [inline services/ui-src/src/measures/globalValidations/tests/validateOneDataSource.test.ts:18]: Can we add a couple other test cases in here as well? Ideally any set of tests should be covering both the expected negative results and positive.

- test for when data source is complete. This should be the default state of the data in `testFormData` and if it's not we should update that seed data to match that state.
- other expected data failures (if DATA_SOURCE.length is 0 but source DATA_SOURCE_SELECTIONS is not, etc)

File: services/ui-src/src/measures/globalValidations/validationsLib.tsx [hunk 2]
```diff
@@ -485,6 +485,18 @@ export const validateRequiredRadioButtonForCombinedRates = (
...
 };
 
+export const validateOneDataSource = (data: Types.DataSource) => {
+  const errorArray: FormError[] = [];
+  if (data?.DataSource?.length === 0) {
+    errorArray.push({
+      errorLocation: "Data Source",
+      errorMessage: "You must select at least one Data Source option",
+    });
+  }
```
Comment (jphechter) [inline services/ui-src/src/measures/globalValidations/validationsLib.tsx:495]: Considering all of our data source strings are constants in `dataConstants.ts`, could we add an interface that further validates what strings we're checking against here? Possibly out of scope for this work, but I think it would be a really cool feature.

**Review by jphechter — CHANGES_REQUESTED** [review]
Would really like to see the tests blown out a little more. Feel free to hit me up.

File: services/ui-src/src/measures/globalValidations/tests/validateOneDataSource.test.ts (hunk 1 above)
Comment (jphechter) [reply services/ui-src/src/measures/globalValidations/tests/validateOneDataSource.test.ts:18]: It could be argued overkill, but the second bullet is addressing what I tried to do in `ensureBothDatesCompletedInRange.test.tsx`. The check isn't just that data is present, but that the shape of the data is as expected.

File: services/ui-src/src/measures/globalValidations/validationsLib.tsx [hunk 3]
```diff
@@ -485,6 +485,18 @@ export const validateRequiredRadioButtonForCombinedRates = (
   return errorArray;
 };
 
+export const validateOneDataSource = (data: Types.DataSource) => {
+  const errorArray: FormError[] = [];
+  if (!data.DataSource || data.DataSource.length === 0) {
```
Comment (jphechter) [inline services/ui-src/src/measures/globalValidations/validationsLib.tsx:490]: @gitsierra @abelzer212 This is more a point of clarification that I didn't consider yesterday. Are we only concerned about the checkboxes in red, or do we also need to validate a selection for the checkboxes in blue? Right now, this function is only concerned about the existence of a top level selection.

I know in the case of EHR or Other we then prompt for a description. Should the user be required to provide a description for one of those selections? This is a pattern we follow in other sections of the measure.

If we don't need to worry about these checks, then this is good to go.

Screen Shot 2022-04-15 at 7 59 20 AM
